lib/github/pull-request.js

```diff
@@ -1,12 +1,20 @@
 const { Project } = require('../models')
 const transformPullRequest = require('../transforms/pull-request')
 const reduceProjectKeys = require('../jira/util/reduce-project-keys')
+const parseSmartCommit = require('../transforms/smart-commit')
 
 module.exports = async (context, jiraClient, util) => {
   const author = await context.github.users.getForUser({ username: context.payload.pull_request.user.login })
   const { data: jiraPayload } = transformPullRequest(context.payload, author.data)
   const { pull_request: pullRequest } = context.payload
 
+  if (!jiraPayload && context.payload.changes) {
+    const hasIssueKeys = !!parseSmartCommit(context.payload.changes.title.from)
+    if (hasIssueKeys) {
+      return jiraClient.devinfo.pullRequest.delete(context.payload.repository.id, pullRequest.number)
```

Comment:
I guess we don't need to specify which issue keys were deleted, right?

Reply:
Nope! It stores the reference in Jira as [`/repository/{repositoryId}/{entityType}/{entityId}`](https://github.com/integrations/jira/blob/master/docs/jira-dev-info-0.10-swagger.yaml#L325), so all we need to do is tell the Jira API that reference is gone and it removes it from the issue.